src/cmd/flux-job.c

```diff
@@ -2854,6 +2862,62 @@ int cmd_wait (optparse_t *p, int argc, char **argv)
     return (rc);
 }
 
+int cmd_annotate (optparse_t *p, int argc, char **argv)
+{
+    int optindex = optparse_option_index (p);
+    flux_t *h;
+    flux_jobid_t id;
+    const char *key;
+    const char *value;
+    void *valbuf = NULL;
+    json_t *v;
+    json_error_t error;
+    json_t *a;
+    flux_future_t *f;
+
+    if ((argc - optindex) < 3) {
+        optparse_print_usage (p);
+        exit (1);
+    }
+    if (!(h = flux_open (NULL, 0)))
+        log_err_exit ("flux_open");
+
+    id = parse_arg_unsigned (argv[optindex], "id");
+    key = argv[optindex + 1];
+
+    if (!strcmp (argv[optindex + 2], "-")) {
+        ssize_t size;
+        if ((size = read_all (STDIN_FILENO, &valbuf)) < 0)
+            log_err_exit ("read_all");
+        value = (const char *)valbuf;
+    }
+    else
+        value = argv[optindex + 2];
```

Comment:
Should exactly argc==3 be required if all remaining args aren't being combined to construct the value?

Also, it would be sort of handy if one could specify a string value without the quotes.